### PR TITLE
Add inital content access identityX route, components & submission handler

### PR DESCRIPTION
**Pull request by B77Mills**
LoggedIn/NonAuthIdentified User:

initial visit:
Screenshot 2024-01-26 at 11 51 03 AM

On submit it will generate a website.ContentAccessSubmission doc containing 
```json
{
    "_id" : ObjectId("65b3e6a21c7736001b111fc7"),
    "payload" : {
        "fields" : {
            "givenName" : {
                "label" : "First Name",
                "value" : "Brian"
            },
            "familyName" : {
                "label" : "Last Name",
                "value" : "Miller"
            },
            ...more
        },
        "selects" : {},
        "booleans" : {}
    },
    "created" : ISODate("2024-01-26T17:06:42.191Z"),
    "contentId" : 22288700,
    "ipAddress" : "10.10.3.1"
}
```

A page reloads occures and the page renders without the gate... Update for discussion is how to handle additional page visits???

Error/success timeout message:
Screenshot 2024-01-26 at 11 45 48 AM

engagement report pr: https://github.com/parameter1/engagement-report/pull/17

### CODE REVIEW

File: services/example-website/server/templates/components/content-body.marko [hunk 1]
```diff
@@ -31,7 +36,48 @@ $ const useIdxDownload = defaultValue(get(content, "gating.surveyType") === "idx
...
+            <marko-web-identity-x-form-access
+              content=content
+              user=identifiedUser
+              formId=idxFormId
+              application=application
+            />
+            $ console.log("Identitfied", identifiedUser);
+          </marko-web-identity-x-non-auth-identify>
+        </else>
+      </marko-web-identity-x-context>
```
Comment (solocommand) [inline services/example-website/server/templates/components/content-body.marko]: Don't create two separate calls and a wrapping component just for a promise. Handle this logic inside the form-access component. For that matter, you don't need the identity-x-context wrapper either, just call your component directly and handle the context promise resolution inside it. The simpler it is to implement, the less the chance of it getting messed up when rolled out.

File: services/example-website/server/templates/components/content-body.marko [hunk 2]
```diff
@@ -31,7 +36,48 @@ $ const useIdxDownload = defaultValue(get(content, "gating.surveyType") === "idx
...
+          <marko-web-identity-x-form-access
+            content=content
+            user=user
+            formId=idxFormId
+            application=application
+          />
+          $ console.log("authenticated", idxFormId);
+        </if>
+        <else>
+          $ console.log('!hasUser')
```
Comment (solocommand) [inline services/example-website/server/templates/components/content-body.marko]: Remove logging -- use `debug` if you plan to log in prod.

File: services/example-website/server/templates/components/content-body.marko [hunk 3]
```diff
@@ -6,7 +6,12 @@ import renderSiteNewsletterMenu from "@parameter1/base-cms-marko-web-theme-monor
 import renderSiteIdxNewsletterMenu from "@parameter1/base-cms-marko-web-theme-monorail/utils/render-site-idx-newsletter-menu";
 
 $ const { content, blockName } = input;
-$ const { contentGatingHandler, contentMeterState, req } = out.global;
+$ const { contentGatingHandler, contentMeterState, contentIdxFormState = {}, req } = out.global;
+
+<!-- update contentIdxFormStateDisplayBase on the req cookies being present -->
+<!-- @todo figure out how to do this ahead of time within the content route if possible -->
+$ contentIdxFormState.displayForm = !Boolean(get(req, `cookies.${get(contentIdxFormState, 'cookie.name')}`));
+$ const { formId: idxFormId, displayForm: displayIdxForm, cookie: idxSuveyCookie } = contentIdxFormState;
```
Comment (solocommand) [inline services/example-website/server/templates/components/content-body.marko:14]: Cleanup -- this logic should not live in the content body for implementing sites. Likely we need to add a new middleware to support generating the access state, and that would need to be added to the monorail theme.

What conditions cause the content to be gated in this manner? How does this interact with IdentityX access level content gating (`canAccess`)?

File: services/example-website/server/templates/components/content-body.marko [hunk 4]
```diff
@@ -68,6 +114,9 @@ $ const useIdxDownload = defaultValue(get(content, "gating.surveyType") === "idx
     />
   </else-if>
   <else>
+    <if(idxFormId)>
```
Comment (solocommand) [inline services/example-website/server/templates/components/content-body.marko]: Empty if?

File: packages/marko-web-identity-x/components/form-access.marko [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      emailConsentRequestEnabled: defaultValue(input.emailConsentRequest, true),
+      regionalConsentPolicies: get(application, "organization.regionalConsentPolicies"),
+    };
+    <marko-web-browser-component name="IdentityXAccess" props=props />
+  </div>
+</if>
+<else>
+  $ const msg = `IdentityX Survey Gating: couldn't load form "${formId}"!`;
+  <p class="text-danger" data-marko-error=msg>
+    $ console.log(msg);
```
Comment (solocommand) [inline packages/marko-web-identity-x/components/form-access.marko]: Remove log. This also shouldn't just display a message, it should throw an error.

**Review by solocommand — CHANGES_REQUESTED** [review]

**Comment by B77Mills** [comment]
[Gated article access & reporting](https://trello.com/c/QH9NAF5Q/508-gated-article-access-reporting)

**Pull request merged**